src/autocomplete.component.js

```diff
@@ -180,6 +193,74 @@
       if (self.clearOnNoSelection && !self.model) {
         clearInput();
       }
+      resetDropDownPostion();
+    }
+
+    /**
+     * Check for drop down position when the element gets focus
+     */
+    function onFocusIn() {
+      resetDropDownPostion();
+
+      if (self.positionTimer) {
+        $timeout.cancel(self.positionTimer);
+      }
+      // Allow plenty of time for the backend to respond to a request before firing
+      self.positionTimer = $timeout(setDropDownPosition, self.positionTimeout * 10);
```

Comment:
When the autocompletes calls the `search` function and we resolve the promise to the backend, we are already calling `setDropDownPosition` through a $timeout. 

A focus-in event will trigger that `search` function above, so do we need to do it again explicitly? Will it not get triggered twice?

Reply:
Not quite - the timer will be cancelled if all OK. If the search function returns an error (or no resuilts) the focus-in will then ensure the drop down is still positioned correctly (even if empty)

Reply:
Cool, wasn't considering the unhappy path.